crates/net/network/src/transactions/fetcher.rs

```diff
@@ -1072,13 +1081,45 @@ impl Default for TransactionFetcher {
     }
 }
 
+#[derive(Debug, Clone)]
+pub(super) struct TxSizeMetadata {
```

Comment:
```suggestion
pub struct TxSizeMetadata {
```

this fixes one warning, please address the other warnings about missing documentation. as for the warning about unused function name, please add an underscore at the start of the method name. I will use the methods when this pr is merged.